src/mon/OSDMonitor.cc

```diff
@@ -8598,8 +8598,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
      CrushWrapper newcrush;
      _get_pending_crush(newcrush);
      if (!newcrush.class_exists(device_class)) {
-       err = -ENOENT;
-       ss << "class '" << device_class << "' does not exist";
+       err = 0; // make command idempotent
```

Comment:
This might need to wait for the commit, since we're testing against newcrush (uncommitted leading edge), not crush (committed state).

Reply:
I would test against crush higher up (and goto reply), and then make this test goto update (or retry).

Reply:
I agree, this will need to either wait or check with the stable map as well.

Say we have a given `foo` class in crush; calling
```
ceph osd class rm foo &
ceph osd class rm foo
```
may have us in a situation such that both commands will act during the same proposal, with one removing class `foo` before the other. The second run can only be returned to the user as if the class has been removed IFF we can ensure it didn't exist to begin with, or once we are sure the proposal has been committed. Otherwise we'll be exposing uncommitted state to the user.